Check improvements

dbendilas opened the pull request:
Make various improvements on how checks work.

## Ignore line length check for URLs
If a line includes a URL (something that has `http://`, `https://` or `ftp://`), this line is not checked for its length. All other lines are checked, as normal. This was introduced because often URLs are very long, and this prevent the line length check of commit messages to pass.

Introduce a configuration option that allows this behaviour: (`ignore_urls`) under `commit_message.body`, default is `True`.

## Add ignore flags for commit messages

Add flag that totally ignores commit message checks (`[!totem]`).
Add flag that ignores a specific line of a commit message body (`#!totem`).

## Allow CLI to run specific checks only
You can define which checks to run.

## Check each commit separately
All checks now can return multiple results, so now each commit is handled as a separate `CheckResult`, thus displaying the information more clearly.

## Allow multiple checks of the same type

The configuration file can now support 2 different structures
for checks, dictionaries and lists.

The new format (list) allows users to include more checks of the same
check type, with different options.

## Fix bug with # of commits in local mode

Before, the local mode only checked the last commit.
Now, all commits up to the first commit of the current branch
are checked.

Commit:
Check each commit separately

All checks now can return multiple results.

Commit:
Allow large URLs in commit message body

Ignore line length if the line includes something that looks like
a URL. The URL doesn't have to be valid.

Introduce a configuration option that allows this behaviour.

Commit:
Fix bug with # of commits in local mode

Before, the local mode only checked the last commit.
Now, all commits up to the first commit of the current branch
are checked.

Commit:
Add ignore flags for commit messages

Add flag that totally ignores commit message checks.
Add flag that ignores a specific line of a commit message body

Commit:
Allow CLI to run specific checks only

Commit:
Add unit tests

Also fix existing unit tests that now failed
because Checks now return multiple results.

wyngarde commented:
# Totem Health Check
Checking if this PR follows the expected quality standards. Powered by [totem](https://www.github.com/transifex/totem).

:white_check_mark: All 4 quality checks have passed! Good job!

Visit the [details page](https://circleci.com/gh/transifex/totem/157) for more information.

Commit:
Add new features in README

Commit:
Allow multiple checks of the same type

The configuration file can now support 2 different structures
for checks, dictionaries and lists.

The new format (list) allows users to include more checks of the same
check type, with different options.

wyngarde commented:
# Totem Health Check
Checking if this PR follows the expected quality standards. Powered by [totem](https://www.github.com/transifex/totem).

:white_check_mark: All 4 quality checks have passed! Good job!

Visit the [details page](https://circleci.com/gh/transifex/totem/161) for more information.
